src/api/db/models/Project.ts

```diff
@@ -0,0 +1,746 @@
+
+      return project;
+    } catch (err) {
+      if (err instanceof GraphQLError) throw err;
+      throw new InternalServerError(err as string);
+    }
+  }
+
+  static async updateProject(
+    input: gql.UpdateProjectInput,
+    context: Context,
+  ): Promise<HydratedProject> {
+    try {
+      const project = await this.queryById(context.user['curr_project']!);
+
+      Object.assign(project, input);
+
+      await project.save();
+
+      return project;
+    } catch (err) {
+      if (err instanceof GraphQLError) throw err;
+      throw new InternalServerError(err as string);
+    }
+  }
+
+  static async createCameraConfig(
+    input: { projectId: string; cameraId: string },
+    context: Context,
+  ): Promise<HydratedProject> {
+    console.log('Project.createCameraConfig - input: ', input);
+    const { projectId, cameraId } = input;
+    try {
+      return await retry(
+        async () => {
+          const [project] = await ProjectModel.getProjects({ _ids: [projectId] }, context);
+          console.log('originalProject: ', project);
+
+          const newCamConfig = {
+            _id: cameraId,
+            deployments: [
+              {
+                _id: new mongoose.Types.ObjectId(),
+                name: 'default',
+                timezone: project.timezone,
+                description: 'This is the default deployment. It is not editable',
+                editable: false,
+              },
+            ],
+          };
+
+          // NOTE: using findOneAndUpdate() with an aggregation pipeline to update
+          // Projects to preserve atomicity of the operation and avoid race conditions
+          // during bulk upload image ingestion.
+          // https://github.com/tnc-ca-geo/animl-api/issues/112
+          const updatedProject = await Project.findOneAndUpdate(
+            { _id: projectId },
+            [
+              { $addFields: { camIds: '$cameraConfigs._id' } },
+              {
+                $set: {
+                  cameraConfigs: {
+                    $cond: {
+                      if: { $in: [cameraId, '$camIds'] },
+                      then: '$cameraConfigs',
+                      else: { $concatArrays: ['$cameraConfigs', [newCamConfig]] },
+                    },
+                  },
+                },
+              },
+            ],
+            { returnDocument: 'after' },
+          );
+
+          console.log('updatedProject: ', updatedProject);
+
+          if (updatedProject!.cameraConfigs.length > project.cameraConfigs.length) {
+            console.log(
+              "Couldn't find a camera config with that _id, so added one to project: ",
+              updatedProject,
+            );
+          }
+
+          return updatedProject!;
+        },
+        { retries: 2 },
+      );
+    } catch (err) {
+      if (err instanceof GraphQLError) throw err;
+      throw new InternalServerError(err as string);
+    }
+  }
+
+  static async createView(
+    input: ViewSchema,
```

Comment:
I think `gql.CreateViewInput` would be more appropriate here, although the properties are identical at the moment. But in the future there's always the potential that they diverge.

Reply:
Fixed